lib/components/term.js

```diff
@@ -146,39 +123,70 @@ export default class Term extends React.PureComponent {
     this.termWrapperRef.appendChild(this.termRef);
 
     if (!props.term) {
+      let needTransparency = Color(props.backgroundColor).alpha() < 1;
+      let useWebGL = false;
+      if (props.webGLRenderer) {
+        if (needTransparency) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            'WebGL Renderer has been disabled since it does not support transparent backgrounds yet. ' +
+              'Falling back to canvas-based rendering.'
+          );
+        } else if (!isWebgl2Supported()) {
+          // eslint-disable-next-line no-console
+          console.warn('WebGL2 is not supported on your machine. Falling back to canvas-based rendering.');
+        } else {
+          // Experimental WebGL renderer needs some more glue-code to make it work on Hyper.
+          // useWebGL = true;
+        }
+      }
+      Term.reportRenderer(props.uid, useWebGL ? 'WebGL' : 'Canvas');
+
       this.term.attachCustomKeyEventHandler(this.keyboardHandler);
+      this.term.loadAddon(this.fitAddon);
+      this.term.loadAddon(this.searchAddon);
+      this.term.loadAddon(new WebLinksAddon());
       this.term.open(this.termRef);
-      this.term.webLinksInit();
-      this.term.winptyCompatInit();
+      if (useWebGL) {
+        this.term.loadAddon(new WebglAddon());
+      }
+    } else {
+      // get the cached plugins
+      this.fitAddon = props.fitAddon;
+      this.search = props.searchAddon;
```

Comment:
I think it should be `this.searchAddon`. I never test the search plugin. you should test them before merge.